src/data.rs

```diff
@@ -60,6 +67,24 @@ impl Data {
             Data::PrivAppendable(ref data) => data.identifier(),
         }
     }
+
+    /// Validate data size.
+    pub fn validate_size(&self) -> bool {
+        let serialised_data = match serialisation::serialise(&self) {
```

Comment:
Ah, sorry, I meant we might want to do _both_: If e.g. the payload is already greater than the limit, there's no need to do the serialisation. If the payload is below the limit, we'd still need to serialise and check.